src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/DefaultEditorTemplates.cs

```diff
@@ -245,6 +248,7 @@ public static IHtmlContent ObjectTemplate(IHtmlHelper htmlHelper)
             }
 
             var serviceProvider = htmlHelper.ViewContext.HttpContext.RequestServices;
+            var htmlEncoder = serviceProvider.GetRequiredService<HtmlEncoder>();
```

Comment:
Should have mentioned 4.
- Move the generally-useful `NullHtmlEncoder` into the Common repo and a new Microsoft.Extensions.NullHtmlEncoder.Sources project / package. This could be similar to the existing Microsoft.Extensions.WebEncoders.Sources package and its `WebEncoders` class: `internal` for most use but `public` for use in the Microsoft.AspNetCore.Razor.Runtime package.